pageserver/src/tenant/timeline.rs

```diff
@@ -2829,28 +2838,26 @@ impl Timeline {
         //
         // TODO: This perhaps should be done in 'flush_frozen_layers', after flushing
         // *all* the layers, to avoid fsyncing the file multiple times.
-        let disk_consistent_lsn = Lsn(lsn_range.end.0 - 1);
-        let old_disk_consistent_lsn = self.disk_consistent_lsn.load();
 
         // If we were able to advance 'disk_consistent_lsn', save it the metadata file.
         // After crash, we will restart WAL streaming and processing from that point.
-        if disk_consistent_lsn != old_disk_consistent_lsn {
-            assert!(disk_consistent_lsn > old_disk_consistent_lsn);
-            self.update_metadata_file(disk_consistent_lsn, layer_paths_to_upload)
+        if let Some(metadata) = metadata {
+            save_metadata(self.conf, &self.tenant_id, &self.timeline_id, &metadata)
                 .await
-                .context("update_metadata_file")?;
+                .context("save_metadata")?;
+
             // Also update the in-memory copy
             self.disk_consistent_lsn.store(disk_consistent_lsn);
         }
```

Comment:
Christian's point about later local update here: https://github.com/neondatabase/neon/pull/5639#pullrequestreview-1694456570

Gc does do:

layer_removal_cs -> write(layermap) -> update_metadata_file
layer_removal_cs -> write(layermap) -> schedule deletions

Perhaps the same could be done in flushing? Seconding however the lack of importance of the local metadata values. Related: the known race #5329.